src/ray/common/ray_object.h

```diff
@@ -63,6 +67,8 @@ class RayObject {
 
   RayObject(rpc::ErrorType error_type);
 
+  RayObject(const ClientID &pinned_at_raylet_id);
```

Comment:
I find it weird that the raylet id (which seems like a global runtime property) is kept as an attribute of RayObject. Doesn't it make more sense to query some global registry to check where the object is pinned?

Reply:
Yeah I actually tried to do this at first in the `ObjectRecoveryManager`, but it turned out into a bit of a headache trying keep it in sync with the ref counter correctly (e.g., we don't want to set the raylet ID if the ref count is already 0). It's nice to keep it in the RayObject because then we can maintain this invariant easily: A plasma object that we own can be fetched <=> the raylet location in the RayObject is set.

Reply:
Not sure I follow. I thought RayObject was a "dummy struct" class, and the pinned location can change. If it changes wouldn't that mean all the existing RayObjects would become invalid?